### ISSUE TITLE
Jepsen tests status

### ISSUE THREAD

**Issue by zamazan4ik** [arangodb/arangodb#16704]
Hi.

During the evaluation of different databases one of the major criteria for us is passing [Jepsen](https://jepsen.io/) test suite. Passing Jepsen test suite confirms some confidence level in testing different quite complicated scenarios in distributed systems.

I did not find any mention of passing Jepsen test suite in the documentation. The only material I have found is an article: https://www.arangodb.com/2017/01/reaching-harnessing-consensus-arangodb/ but seems like it is kinda outdated.

My questions are:

- What is the current state of Jepsen tests for the database?
- Do you plan to invest your time into the Jepsen tests? Maybe your current test suite covers all Jepsen cases (or maybe perform even more checks)?
- If the database pass (or not pass) Jepsen test suite, would be awesome if you put this information somewhere into the documentaion. It is really important for the end-users.

Thanks in advance!

**Comment by MBkkt** [comment]
Hi, your issue is really strange.

Don't exist some common for all databases jepsen test suite.
It should be written by someone, it can be written incorrect, etc...

If you see Jepsen reports, it's Kyle and other devs commonly not for free, test concrete databases.

We used jepsen framework to test our raft implementation (Agency), but it's probably outdated.
Maybe we will update it sometime.

> What is the current state of Jepsen tests for the database?

We don't use Jepsen framework for tests now.

> Do you plan to invest your time into the Jepsen tests? Maybe your current test suite covers all Jepsen cases (or maybe perform even more checks)?

As I wrote, no exist some Jepsen cases, exist Jepsen framework, with a lot of fault injection settings, and user can write tests for it.

Maybe sometime we will add it, but in general I don't like Jepsen framework, it's probably intentional written on Closure.
So personally I prefer native frameworks for fault injection, like in FoundationDB for example.
Another work in this direction is https://developers.facebook.com/blog/post/2022/11/22/hermit-deterministic-linux-testing/

**Comment by zamazan4ik** [comment]
Yes, you are right. I mostly meant introducing some tests for fault-tolerance in "Jepsen" style. I have chosen not clear enough description in the start message - my bad. Honestly, as a user, I do not care much about the implementation of fault-tolerance testing of a database - I just want to check that the database has a mature fault-tolerance test suite and is tested for the fault-tolerance on a regular basis.

**Comment by MBkkt** [comment]
I understood, sorry but now we only have manual fault injection in our regular tests
